service/controller/v14/resource/bridgezone/resource.go

```diff
@@ -388,7 +388,7 @@ func (r *Resource) getNameServersAndTTL(ctx context.Context, client *route53.Rou
 	rs := *out.ResourceRecordSets[0]
 
 	if strings.TrimSuffix(*rs.Name, ".") != name {
-		return nil, 0, microerror.Maskf(executionError, "expected NS recrod with name %q , found %q", name, *rs.Name)
+		return nil, 0, microerror.Maskf(notFoundError, "NS recrod %q for HostedZone %q not found", name, zoneID)
```

Comment:
This works well when there is only one or zero delegations. Pretty hard to find in the test env.

Reply:
Nitpick and already there but would be nice to fix the typo on `record`.